### PR TITLE
feat: add MLH fellowship to open source programs

### LINKED ISSUE #2134: [Add] MLH Link under open source programs
### Site URL

https://fellowship.mlh.io/

### Description of the above provided site

Major League Hacking is a popular mentorship program that allows you to contribute to widely-used open source projects.

### Category

Open Source

### Sub Category

open-source

### Checklist

- [X] I have checked the existing issues
- [X] I have read the [Contributing Guidelines](https://github.com/rupali-codes/LinksHub/blob/main/CONTRIBUTING.md)
- [X] I am willing to work on this issue (optional)
- [X] I am participating in Hacktoberfest 2023

### PR DESCRIPTION
**Pull request by vg300**
## Fixes Issue: 

closes #2134  

## Changes proposed
This PR adds MLH to the open source programs category.

### CODE REVIEW

**Comment by CBID2** [comment]
Hi @vg300. Welcome to LinksHub! :) We don't accept spam pull requests, so I'm putting yours in draft mode for now. In the future, please raise an issue and wait until it has been approved and assigned to you before you submit a pull request. Refer to our [Contributing Guide](https://github.com/rupali-codes/LinksHub/blob/main/CONTRIBUTING.md) to learn how to effectively contribute to the project. Happy Contributing! 😊

File: database/open_source/os-programs.json [hunk 1]
```diff
@@ -33,5 +33,13 @@
       "url": "https://events.linuxfoundation.org/about/calendar/",
       "category": "open-source",
       "subcategory": "os-programs"
+    },
+    {
+      "name": "MLH Fellowship",
+      "description": "The MLH Fellowship is a 12-week internship for aspiring software engineers to contribute to the type of Open Source projects that every company depends on.",
```
Comment (CBID2) [inline database/open_source/os-programs.json:39]:
```suggestion
      "description": "The MLH Fellowship is a 12-week internship for aspiring software engineers to contribute to Open Source projects that companies depend on.",
```

**Review by CBID2 — CHANGES_REQUESTED** [review]
You're off to a good start @vg300. I gave a suggestion on how to improve it a bit.

**Comment by vg300** [comment]
Hi @CBID2, thanks for your review. I have incorporated your suggestion. Please let me know if anything else is required :)

**Review by CBID2 — APPROVED** [review]
Looks good to me. @Anmol-Baranwal, your turn 😊

**Review by Anmol-Baranwal — APPROVED** [review]
🚀 Looks good to me.

**Review by rupali-codes — APPROVED** [review]

**Pull request merged**